### PR TITLE
OCPBUGS-7092 SiteConfig crTemplates override does not work with a ConfigMap CR

### PR DESCRIPTION
**Pull request by serngawy**
SiteConfig crTemplates override does not work with a ConfigMap CR

### CODE REVIEW

File: ztp/siteconfig-generator/siteConfig/testdata/ConfigMapOverride.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Extra manifest will be added to the data section
+kind: ConfigMap
+apiVersion: v1
+metadata:
+    annotations:
+        argocd.argoproj.io/sync-wave: "1"
+        overrideconfigmap-annotation/test: "new-annotation"
+    name: "{{ .Cluster.ClusterName }}"
+    namespace: "{{ .Cluster.ClusterName }}"
+data:
```
Comment (lack) [inline ztp/siteconfig-generator/siteConfig/testdata/ConfigMapOverride.yaml:10]: To make this valid, should we maybe have this point at an empty map:
```suggestion
data: {}
```

Comment (serngawy) [reply ztp/siteconfig-generator/siteConfig/testdata/ConfigMapOverride.yaml]: done

**Pull request merged**